lib/modules/datasource/docker/index.ts

```diff
@@ -824,41 +824,86 @@ export class DockerDatasource extends Datasource {
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
+        newValue ?? 'undefined'
+      })`
     );
     const newTag = newValue ?? 'latest';
     let digest: string | null = null;
     try {
+      let architecture: string | null = null;
+      if (currentValue && currentDigest) {
+        const manifestResponse = await this.getManifestResponse(
+          registryHost,
+          dockerRepository,
+          currentValue
+        );
+        if (manifestResponse) {
+          const manifestList = JSON.parse(manifestResponse.body) as
+            | ImageList
+            | OciImageList;
+          for (const manifest of manifestList.manifests) {
+            if (manifest.digest === currentDigest) {
+              architecture =
+                (manifest.platform['architecture'] as string) || null;
+              logger.debug(
+                `Current digest ${currentDigest} relates to architecture ${
+                  architecture ?? 'null'
+                }`
+              );
+              break;
+            }
+          }
+        }
+      }
+
       let manifestResponse = await this.getManifestResponse(
         registryHost,
         dockerRepository,
         newTag,
         'head'
       );
       if (manifestResponse) {
-        if (hasKey('docker-content-digest', manifestResponse.headers)) {
+        if (
+          !architecture &&
+          hasKey('docker-content-digest', manifestResponse.headers)
+        ) {
           digest =
             (manifestResponse.headers['docker-content-digest'] as string) ||
             null;
         } else {
           logger.debug(
             { registryHost },
-            'Missing docker content digest header, pulling full manifest'
+            'Architecture-specific digest or missing docker-content-digest header - pulling full manifest'
           );
           manifestResponse = await this.getManifestResponse(
             registryHost,
             dockerRepository,
             newTag
           );
-          digest = extractDigestFromResponseBody(manifestResponse!);
+          if (architecture && manifestResponse) {
```

Comment:
Move thise before the `hasKey('docker-content-digest', manifestResponse.headers)` condition

Reply:
Won't work because the `HEAD` request won't return the body needed to get the architecture-specific digest. Anyway, I have now simplified / optimized the request flow so that a HEAD request is omitted for architecture digest, as we always need to the full manifest for platform digests.

Summarizing, the most-recent version now supports these two workflows:
- manifest digest case (_same as until now, no tests changed_):
  - HEAD request -> digest from header or if not found -> full body request -> digest from full manifest body
- architecture digest case (_new with this PR_):
  - full body request -> architecture-specific digest or fallback to digest from full manifest body if not found